GUI/MainRepo.cs

```diff
@@ -64,7 +66,22 @@ private void UpdateRepo(object sender, DoWorkEventArgs e)
             try
             {
                 AddStatusMessage("Updating repositories...");
-                e.Result = Repo.UpdateAllRepositories(RegistryManager.Instance(CurrentInstance), CurrentInstance, Manager.Cache, GUI.user);
+
+                // Note the current mods' compatibility for the NewlyCompatible filter
+                KspVersionCriteria versionCriteria = CurrentInstance.VersionCriteria();
+                IRegistryQuerier registry = RegistryManager.Instance(CurrentInstance).registry;
+                Dictionary<string, bool> oldModules = registry.Available(versionCriteria)
+                    .ToDictionary(m => m.identifier, m => false);
+                registry.Incompatible(versionCriteria)
+                    .Where(m => !oldModules.ContainsKey(m.identifier))
```

Comment:
Can mods be in `registry.Incompatible` but not in `registry.Available`?

Reply:
Yes, that's the idea. The two functions together return non-overlapping sets, the union of which is all modules (see #2216).

https://github.com/KSP-CKAN/CKAN/blob/5331bf9344f6729a6b96f52bea45bdec245a3bb3/Core/Registry/Registry.cs#L476-L526

Reply:
Oh I see where my confusion comes from. `Registry.Available` basically means `Registry.Compatible` and only includes the latest compatible version of all mods,
contrary to `Registry.available_modules` which includes all mods and mod versions and is saved in the registry.json?
Seems the naming scheme is a bit mixed up.

Reply:
Yup. I started a branch to rename Available to Compatible, but there are complications, such as the `ckan available` command only listing compatible modules. It's not a simple matter to reach a consistent state.